src/views/RoomDetailsView.vue

```diff
@@ -0,0 +1,154 @@
+<script>
+
+import axios from "axios";
+import {FontAwesomeIcon} from "@fortawesome/vue-fontawesome";
+import {ref} from "vue";
+
+export default {
+  name: "RoomDetailsView",
+  components: {FontAwesomeIcon},
+
+  data() {
+    return {
+      room: null,
+      roomId: null,
+      currentImageIndex: ref(0),
+      // Assuming room details are fetched and stored in `room`
+      imageSources: [],
+
+    }
+  },
+
+  watch: {
+    async room(newRoom) {
+      if (newRoom && newRoom.id) {
+        try {
+          const imagePath = await import(`@/assets/images/rooms/${newRoom.id}.jpg`);
+          this.imageSources = [imagePath.default];
+        } catch (error) {
+          console.error('Error loading image:', error);
+          // Handle the error or set a default image
+        }
+      }
+    }
+  },
+
+
+  computed: {
+    imagePath() {
+      return this.imageSources.length > 0 ? this.imageSources[this.currentImageIndex] : '';
+    }
+  },
+
+
+  created() {
+    this.roomId = this.$route.params.roomId;
+    this.fetchRoomDetails();
+  },
+
+  methods: {
+    fetchRoomDetails() {
+      this.roomId = this.$route.params.roomId
+      console.log(this.roomId)
+
+      axios.get(`https://boutique-hotel.helmuth-lammer.at/api/v1/rooms/${this.roomId}`)
```

Comment:
Soll in eigenem Store gefetcht werden (Pinia)

Reply:
wird in eigenem Branch erledigt